xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingService.java

```diff
@@ -5,66 +5,82 @@
-    if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
-      return messageType + ":" + message.get("marketId").asText();
+    @Override
+    protected String getChannelNameFromMessage(JsonNode message) {
+        final String messageType = message.get("messageType").asText();
+        if (messageType.startsWith(CHANNEL_TICK)) {
+            return messageType + ":" + message.get("marketId").asText();
+        } else if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
+            return messageType + ":" + message.get("marketId").asText();
+        } else
+            return messageType;
     }
-    return messageType;
-  }
 
-  @Override
-  public String getSubscribeMessage(String channelName, Object... args) throws IOException {
-    if (CHANNEL_ORDERBOOK.equals(channelName)) {
-      subscribedOrderbooks.add(args[0].toString());
-      LOG.debug("Now subscribed to orderbooks {}", subscribedOrderbooks);
-      return objectMapper.writeValueAsString(buildSubscribeMessage());
-    } else {
-      throw new IllegalArgumentException(
-          "Can't create subscribe messsage for channel " + channelName);
+    @Override
+    public String getSubscribeMessage(String channelName, Object... args) throws IOException {
+        if (CHANNEL_TICK.equals(channelName)) {
+            subscribedtick.add(args[0].toString());
+            LOG.debug("Now subscribed to Ticker {}", subscribedtick);
+            return objectMapper.writeValueAsString(buildSubscribeMessage());
+        } else if (CHANNEL_ORDERBOOK.equals(channelName)) {
+            subscribedOrderbooks.add(args[0].toString());
+            LOG.debug("Now subscribed to Orderbook {}", subscribedOrderbooks);
+            return objectMapper.writeValueAsString(buildSubscribeMessage());
+        } else {
+            throw new IllegalArgumentException(
+                    "Can't create subscribe messsage for channel " + channelName);
+        }
     }
-  }
 
-  public String getSubscriptionUniqueId(String channelName, Object... args) {
-    if (CHANNEL_ORDERBOOK.equals(channelName)) {
-      return channelName + ":" + args[0].toString();
+    public String getSubscriptionUniqueId(String channelName, Object... args) {
+        if (CHANNEL_TICK.equals(channelName)) {
+            return channelName + ":" + args[0].toString();
+        } else if (CHANNEL_ORDERBOOK.equals(channelName)) {
+            return channelName + ":" + args[0].toString();
+        } else {
+            return channelName;
+        }
     }
-    return channelName;
-  }
 
-  @Override
-  public String getUnsubscribeMessage(String channelName) throws IOException {
-    if (channelName.startsWith(CHANNEL_ORDERBOOK)) {
-      subscribedOrderbooks.remove(channelName);
-      return objectMapper.writeValueAsString(buildSubscribeMessage());
-    } else {
-      return null;
+    @Override
+    public String getUnsubscribeMessage(String channelName) throws IOException {
+        if (channelName.startsWith(CHANNEL_TICK)) {
```

Comment:
I think you're going to need a proper `BTCMarketsWebSocketUnsubscribeMessage` here given that the `messageType` is subscribe in `BTCMarketsWebSocketSubscribeMessage`. I don't see how this would unsubscribe?

Reply:
Done